src/components/Popstate.js

```diff
@@ -0,0 +1,20 @@
+let setup = false
+export const Popstate = ({ update }) => {
+  if (!setup) {
+    setup = true
+    // Update path when a pushState occurs
+    const pushState = history.pushState
+    history.pushState = function() {
```

Comment:
Can these `history` modifications create conflicts with other libraries or unintended bugs?

This is a pretty clever trick to get around passing the `go` function around.

Reply:
There has been discussion about this on slack too (you really should join us!) and as you can see, it leaves `pushState` and `popState` exactly as is, but extends it (ensures an action is called subsequently). The only real risk I can see here is if the action for some reason ceases to exist.. to guard agains that `typeof update === 'function' && update()` was proposed.